lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/EagerFallback.cpp

```diff
@@ -175,25 +198,34 @@ void eager_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack,
                       "mutable alias: ",
                       schema_returns[idx]);
         } else {
+          c10::optional<c10::Device> tgt_device = compute_target_device(tensor_args, tensorlist_args);
           if (alias_info.has_value() && !alias_info.value().isWrite()) {
             // immutable alias (view) case: Warn here, since we're copying and
             // not creating a view.
             // If this operator is needed, the backend should provide a kernel
             // for it.
             // See Note [Eager Fallback Does Not Handle View Operators]
-            auto tgt_device = tensor_args[0].device();
+            std::stringstream dev_str;
+            if (tgt_device) {
+                dev_str << *tgt_device;
+            } else {
+                dev_str << "<none>";
+            }
             TORCH_WARN(false, "The operator ", op.schema().operator_name(),
                        " appears to be a view operator, ",
                        "but it has no implementation for the backend \"",
-                       tgt_device, "\". View operators don't support ",
+                       dev_str.str(), "\". View operators don't support ",
                        "falling back to run on the eager, since the tensor's "
                        "storage cannot be shared across devices.");
           }
           // Case (2): copy case. Copy the eager output tensor to the original
           // device.
-          auto tgt_device = tensor_args[0].device();
-          (*stack)[returns_begin + idx] =
-              c10::IValue(returns[idx].toTensor().to(tgt_device));
+
+          // We technically  might not have a target device, e.g. if you call torch.cat() with an empty list
+          // In that case, we shouldn't have any tensors to schlep across devices anyway.
+          if (tgt_device) {
+              (*stack)[returns_begin + idx] = c10::IValue(returns[idx].toTensor().to(*tgt_device));
```

Comment:
so was this the place where it was segfaulting previously, and the majority of the change is to recognize the cases where tgt_device may be unavailable?

Reply:
The main problem was `cat` takes `TensorList` as its first input instead of `Tensor`. https://github.com/pytorch/pytorch/pull/60993 made one step further by assuming the input list could be empty.